tests/integration/cmor/_fixes/cmip6/test_fgoals_g3.py

```diff
@@ -95,3 +98,46 @@ def test_get_siconc_fix():
 def test_siconc_fix():
     """Test fix for ``siconc``."""
     assert Siconc is Tos
+
+
+def test_get_mrsos_fix():
+    """Test getting of fix."""
+    fix = Fix.get_fixes('CMIP6', 'FGOALS-g3', 'Lmon', 'mrsos')
+    assert fix == [Mrsos(None)]
+
+
+def test_mrsos_fix():
+    """Test fix for ``mrsos``."""
+    assert issubclass(Mrsos, Fix)
+
+
+@mock.patch('esmvalcore.cmor._fixes.cmip6.fgoals_g3.Fix.fix_metadata',
+            autospec=True)
+def test_mrsos_fix_metadata(mock_base_fix_metadata):
+    """Test ``fix_metadata`` for ``mrsos``."""
+    mock_base_fix_metadata.side_effect = lambda x, y: y
+
+    # Create test cube
+    lat_coord = iris.coords.AuxCoord([1.0, 2.0, 3.0],
+                                     var_name='lat',
+                                     standard_name='latitude')
+    lat_coord.bounds = [[0.5, 1.5], [-0.5, 0.5], [2.5, 3.5]]
+    lon_coord = iris.coords.AuxCoord([1.0, 2.0, 3.0],
+                                     var_name='lon',
+                                     standard_name='longitude')
+    lon_coord.bounds = [[0.5, 1.5], [-0.5, 0.5], [2.5, 3.5]]
+    cube = iris.cube.Cube([1.0, 2.0, 3.0],
+                          var_name='mrsos',
+                          standard_name='mass_content_of_water_in_soil_layer',
+                          aux_coords_and_dims=[(lat_coord, 0), (lon_coord, 0)])
+    cubes = iris.cube.CubeList([cube])
+
+    # Apply fix
+    vardef = get_var_info('CMIP6', 'Lmon', 'mrsos')
+    fix = Mrsos(vardef)
+    fixed_cubes = fix.fix_metadata(cubes)
+    assert len(fixed_cubes) == 1
+    fixed_cube = fixed_cubes[0]
+    assert _check_bounds_monotonicity(fixed_cube.coord('latitude'))
+    assert _check_bounds_monotonicity(fixed_cube.coord('longitude'))
```

Comment:
You are checking the code by calling parts of the same code. That's bad practice, as some problems in the `_check_bounds_monotonicity` can go unnoticed

In fact, the check should be easier than that, something like

```python
np.testing.assert_allclose(
        fixed_cube.coord('latitude').bounds, 
       [[ 0.5, 1.5], [1.5, 2.5], [2.5, 3.5]]
)
```

By the way, it would have been ok if there was a separate test for the `_check_bounds_monotonicity` function, but that would have been a complete overkill as we are just interested in checking if the data is correctly fixed.

Reply:
Thanks, that makes sense. I have modified the test to use your suggestion.